### PR TITLE
Cleanup helix health report

### PR DESCRIPTION
**Pull request by justinlin-linkedin**
This PR would clean up some legacy code for helix health report
We don't report the storage stats to helix as health report anymore and we shouldn't do that even in code.

### CODE REVIEW

**Review by zzmao — COMMENTED** [review]
LGTM

File: ambry-server/src/main/java/com/github/ambry/server/AmbryServer.java [hunk 1]
```diff
@@ -312,17 +312,16 @@ public void startup() throws InstantiationException {
       }
 
       // Other code
-      List<AmbryHealthReport> ambryHealthReports = new ArrayList<>();
+      List<AmbryStatsReport> ambryStatsReports = new ArrayList<>();
       Set<String> validStatsTypes = new HashSet<>();
       for (StatsReportType type : StatsReportType.values()) {
         validStatsTypes.add(type.toString());
       }
       if (serverConfig.serverStatsPublishHealthReportEnabled) {
```
Comment (SophieGuo410) [inline ambry-server/src/main/java/com/github/ambry/server/AmbryServer.java:320]: nit: you may wants to rename the config file to serverStatsPublishStatsReportEnabled?

File: ambry-server/src/main/java/com/github/ambry/server/AmbryStatsReportImpl.java [hunk 2]
```diff
@@ -14,53 +14,27 @@
 
 package com.github.ambry.server;
 
-import java.util.HashMap;
-import java.util.Map;
-import org.apache.helix.healthcheck.HealthReportProvider;
-
-
 /**
  * Customized Helix Health Report for collecting account usage stats per node (instance).
```
Comment (SophieGuo410) [inline ambry-server/src/main/java/com/github/ambry/server/AmbryStatsReportImpl.java:18]: Update the comments correspondingly?

**Review by SophieGuo410 — COMMENTED** [review]
Overall lgtm, left some minor comments.

**Review by SophieGuo410 — APPROVED** [review]
lgtm.

**Pull request merged**